File: app/src/main/java/by/nts/cafe/app/presentation/ui/tables/menu/AddDishDialogFragment.java
package by.nts.cafe.app.presentation.ui.tables.menu;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.TabLayout;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.ImageButton;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import by.nts.cafe.app.R;
import by.nts.cafe.app.model.DishModel;

public class AddDishDialogFragment extends DialogFragment implements MenuFragment.OnDishSelectedListener {

    //region Declarations
    @BindView(R.id.tabs)
    TabLayout tabLayout;
    @BindView(R.id.viewPager)
    ViewPager viewPager;
    @BindView(R.id.blockAmount)
    View blockAmount;
    @BindView(R.id.txtDishName)
    TextView txtDishName;
    @BindView(R.id.btnPlus)
    ImageButton btnPlus;
    @BindView(R.id.btnMinus)
    ImageButton btnMinus;
    @BindView(R.id.fabAddDish)
    FloatingActionButton fabAddDish;

    private DishModel dishSelected;
    //endregion

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setStyle(DialogFragment.STYLE_NO_FRAME, R.style.AppTheme_DishDialog);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.dialog_dish_add, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        ButterKnife.bind(this, view);

        viewPager.setAdapter(new FragmentPagerAdapter(getChildFragmentManager()) {
            @Override
            public Fragment getItem(int position) {
                return new MenuFragment();
            }

            @Override
            public int getCount() {
                return 6;
            }
        });
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(viewPager));

        tabLayout.addTab(tabLayout.newTab().setText("First"));
        tabLayout.addTab(tabLayout.newTab().setText("Soups"));
        tabLayout.addTab(tabLayout.newTab().setText("Drinks"));
        tabLayout.addTab(tabLayout.newTab().setText("Alcohol"));
        tabLayout.addTab(tabLayout.newTab().setText("Deserts"));
        tabLayout.addTab(tabLayout.newTab().setText("Misc."));
    }

    @Override
    public void onStart() {
        super.onStart();

        if (getDialog() != null) {
            getDialog().setCancelable(true);
            getDialog().setCanceledOnTouchOutside(true);
            getDialog().getWindow().setLayout(getContext().getResources().getDimensionPixelOffset(R.dimen.dialog_dish_width),
                    WindowManager.LayoutParams.MATCH_PARENT);
        }
    }

    @OnClick(R.id.fabAddDish)
    protected void addDishToOrder() {
        // todo call for presenter

        dishSelected = null; // call when saved and hide the block
    }

    @Override
    public void onDishSelect(DishModel dishModel) {
        dishSelected = dishModel;
        blockAmount.setVisibility(View.VISIBLE);
        txtDishName.setText(dishModel.getName());
    }
}

File: app/src/main/java/by/nts/cafe/app/presentation/ui/menu/AddDishDialogFragment.kt
package by.nts.cafe.app.presentation.ui.menu

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.DialogFragment
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import by.nts.cafe.app.R
import by.nts.cafe.app.model.db.DishModel
import kotlinx.android.synthetic.main.dialog_dish_add.*

class AddDishDialogFragment : DialogFragment(), MenuFragment.OnDishSelectedListener {

    //region Declarations
    private var dishSelected: DishModel? = null
    //endregion

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setStyle(DialogFragment.STYLE_NO_FRAME, R.style.AppTheme_DishDialog)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.dialog_dish_add, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)


        viewPager.adapter = object : FragmentPagerAdapter(childFragmentManager) {
            override fun getItem(position: Int): Fragment {
                return MenuFragment()
            }

            override fun getCount(): Int {
                return 6
            }
        }
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabs))
        tabs.addOnTabSelectedListener(TabLayout.ViewPagerOnTabSelectedListener(viewPager))

        tabs.addTab(tabs.newTab().setText("First"))
        tabs.addTab(tabs.newTab().setText("Soups"))
        tabs.addTab(tabs.newTab().setText("Drinks"))
        tabs.addTab(tabs.newTab().setText("Alcohol"))
        tabs.addTab(tabs.newTab().setText("Deserts"))
        tabs.addTab(tabs.newTab().setText("Misc."))
        fabAddDish.setOnClickListener { addDishToOrder() }
    }

    override fun onStart() {
        super.onStart()

        if (dialog != null) {
            dialog.setCancelable(true)
            dialog.setCanceledOnTouchOutside(true)
            dialog.window!!.setLayout(context!!.resources.getDimensionPixelOffset(R.dimen.dialog_dish_width),
                    WindowManager.LayoutParams.MATCH_PARENT)
        }
    }

    fun addDishToOrder() {
        // todo call for presenter

        dishSelected = null // call when saved and hide the block
    }

    override fun onDishSelect(dishModel: DishModel) {
        dishSelected = dishModel
        blockAmount!!.visibility = View.VISIBLE
        txtDishName!!.text = dishModel.name
    }
}
